### PR TITLE
Bump com.gradle.plugin-publish from 0.21.0 to 1.0.0

### PR DESCRIPTION
**Pull request by erichaagdev**
Dependabot was unable to perform an automated update of `com.gradle.plugin-publish` from 0.21.0 to 1.0.0. 

This PR updates the plugin and migrates to the new API.

### CODE REVIEW

**Review by etiennestuder — APPROVED** [review]
LGTM. Let's wait for the review from Alexis before merging.

**Comment by runningcode** [comment]
@alextu could you take a quick look at this?

**Comment by alextu** [comment]
Sorry for the late response, looks good. Here's a similar PR I did for the CCUD: https://github.com/gradle/common-custom-user-data-gradle-plugin/pull/69/files, only difference is about signing, which is not important here since it's not a Gradle plugin published to the Gradle portal

**Pull request merged**
